Club house project

Leon-Mbegera opened the pull request:
This is a clubhouse project where members or its users can write anonymous posts.
- Only if a member is signed in is when they can see the author of a post, otherwise, they will only see the title and context of the post.
- It includes links for new members to sign up, links to sign in, and another link to sign out.

Leon-Mbegera committed:
Add config files

Leon-Mbegera committed:
Modify README.md file

Leon-Mbegera committed:
Install devise gem and set up default URL options for devise mailer in each environment

Leon-Mbegera committed:
Generate devise User

Leon-Mbegera committed:
Add username field to users table

Leon-Mbegera committed:
Controller filters and helpers

Leon-Mbegera committed:
Generate Post model and Posts controller

Leon-Mbegera committed:
Create corresponding resource in routes for new, create and index actions

Leon-Mbegera committed:
Restrict access to new and create action to only signed in users

Leon-Mbegera committed:
Prepare new action in post controller

Leon-Mbegera committed:
Create form in new view to create new post

Leon-Mbegera committed:
Make create action to build a post

Leon-Mbegera committed:
Add index action to post controller

Leon-Mbegera committed:
Create view for index action for list of posts

Leon-Mbegera committed:
Add logic to display author to signed in users

Leon-Mbegera committed:
Fix syntactical errors in files

Leon-Mbegera committed:
Add validation check for title and body

Leon-Mbegera committed:
Generate devise for views

Leon-Mbegera committed:
Correct rubocop offenses

SajjadAhmad14 commented on app/views/devise/passwords/edit.html.erb:
```diff
@@ -0,0 +1,25 @@
+
+  <div class="field">
+    <%= f.label :password, "New password" %><br />
+    <% if @minimum_password_length %>
+      <em>(<%= @minimum_password_length %> characters minimum)</em><br />
+    <% end %>
```
Please create a helper under app/helpers and move this logic there. [OPTIONAL]

SajjadAhmad14 reviewed (changes requested):
# Changes Required :recycle:
> "Change is inevitable. Change is constant. -Benjamin Disraeli"
## Hello, you have been doing a tremendous job so far :ok_hand:
## To Highlight 🎉
- [x] README has all the instructions to run the project locally.
- [x] Commit messages are descriptive.
## To Improve 🙏
- [x] Please use proper Git flow. The main/master branch should be there which is used for production releases.
- [x] PR should have a descriptive summary.
- [x] User should be able to sign in. Please see this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/sign_in_requirement.png).
- [x] A signed-in user you can create a new post. Please see this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/signed_in_can_post_requirement.png)
- [x] A signed-in user can see the post's author. Please see this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/signed_in_can_see_author_requirement.png).
- [x] User should be able to sign out. Please see this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/sign_out_requirement.png).
- [x] A guest user can only see the post's text. Please see this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/guest_can_see_only_posts_requirement.png).
- [x] Sign in/Sign out links should be present in any view of the app. Please see this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/sign_in_out_link_requirement.png).
- [x] Posts controller should use #before_action to :new and :create. Please see this [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/authentication/before_action_requirement.png).
- [x] Project does not have Users controller. Please make sure it has a controller in order to authenticate the user.
- [x] App has no functionality as per requirements. The app shows only this page:
![issues](https://user-images.githubusercontent.com/35504149/122596244-ffbf8700-d082-11eb-9a23-242ec5834b68.png)

- [x] Please see inline comments for reference.
# Optional
- [x] Views should be completely separate from app logic. You can use app/helpers to move all of your view-related logic.

### Happy Coding ☺️

## If you have any questions you can comment here in this PR or can always connect with me on slack ( @Sajjad Ahmad ).

Leon-Mbegera committed:
Modify posts show page

Leon-Mbegera committed:
Modify posts view

Leon-Mbegera committed:
Create sign up, sign in and sign out links

Leon-Mbegera committed:
Add authentication for users to sign up before creating posts

Leon-Mbegera committed:
Generate Users controller

Leon-Mbegera committed:
Modify to login link to head user to login form

Leon-Mbegera committed:
Correct rubocop offenses

Leon-Mbegera commented:
I have addressed all the requested changes herein,
Thanks for the straightforward guidelines on what to change.
gracias`

george-shammar reviewed (changes requested):
### **STATUS:CHANGES REQUIRED** :recycle: :recycle: :construction: :construction_worker_man: 

Decent effort on this project. :+1: 

**Highlight**

- [x] Thank you for using the correct gitflow and implementing some of the required changes.

Please see required changes below:

- [x] As a signed-in user, you can create a new post. There is no where on your page to create a new post without manually editing the url which is not ideal.
- [x] A signed-in user should be able to see the post's author. But if posts cannot be created, it can't be seen.
- [x] A guest user should only see the post's text. Same reason as above - post cannot be created.
- [x] Sign in/Sign out links should be present in any view of the app. There is no way to link other pages of the application. 
- [ ] Project does not have Users controller. In trying to implement this, you have created a huge "User" directory and commented out every code inside all the files. Please get rid of all commented unused codes.

Please make the required changes and submit another CRR when you're ready.

In need of clarification or support, please drop your message here and ping me on slack.

Cheers and happy coding. :+1: :smile:

Leon-Mbegera committed:
Add link to create new form page

Leon-Mbegera committed:
Add authentication for only signed in members to create a post

Leon-Mbegera committed:
Make sign up, sign in and sign out links on every page

Leon-Mbegera committed:
Remove all commented unused code in user controller

Leon-Mbegera commented:
Herein, I have addressed the requested changes:

1. As a signed user, he/she can create a post
2. Only a signed-in user can see the author of a post
3. A guest user can only see the post text
4. I've displayed sign up/sign in, and sign out links on every page. Possible to navigate to other pages
5. I've deleted all unused code in the user controller.

SajjadAhmad14 reviewed (approved):
# Status: Approved :heavy_check_mark: :heavy_check_mark: :heavy_check_mark:

## Congratulations 🎉 Thank you for implementing changes suggested previously 🙏
# OPTIONAL
- [x] You can show flash messages for success and failure for user-friendliness.
- [x] Views should be completely separate from app logic. Please move view-related logic under app/helpers.
### Happy Coding ☺️

## If you have any questions you can comment here in this PR or can always connect with me on slack ( @Sajjad Ahmad ).
